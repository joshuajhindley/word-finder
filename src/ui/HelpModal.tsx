import React from 'react'
import '../styles/Finder.scss'
import ReactModal from 'react-modal'
import classNames from 'classnames'

export default class HelpModal extends React.Component<any, any> {
  render() {
    const { darkMode, showHelp, actions } = this.props

    const colorDiv = classNames({
      dark: darkMode
    })

    return (
      <ReactModal
        isOpen={showHelp}
        onRequestClose={() => actions.toggleHelp(false)}
        ariaHideApp={false}
        overlayClassName='modal-overlay'
        className={`help-modal ${colorDiv}`}
      >
        <div className='modal-content'>
          <div className='help-modal-header'>
            <div className='help-modal-header-title'>
              <h1>Help</h1>
            </div>
            <div className='help-modal-header-close' onClick={() => actions.toggleHelp(false)}>
              <h1>x</h1>
            </div>
          </div>
          <div className='help-modal-body'>
            <p>
              This is a simple tool to help you find the correct word in a Wordle game.
              <br />
              <br />
              There are three rows of input boxes.
              <br />
              The first row are letters in the correct position.
              <br />
              These are the green letters in your Wordle game.
              <br />
              <br />
              The second row are letters that are in the final word, but are in the wrong position.
              <br />
              These are the yellow letters in your Wordle game.
              <br />
              <br />
              The third row are letters that are not in the word.
              <br />
              These are the grey letters in your Wordle game.
              <br />
              New input boxes are added to this row automatically as you add letters.
              <br />
              <br />
              You can use the 'search' button to find the first 500 words that match the letters you've entered and can
              reset the input boxes by tapping the 'reset' button.
            </p>
          </div>
        </div>
      </ReactModal>
    )
  }
}
