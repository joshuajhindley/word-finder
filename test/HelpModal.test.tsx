import React from 'react'
import { mount, shallow } from 'enzyme'
import HelpModal from '../src/ui/HelpModal'

describe('tests the HelpModal component', () => {
  const actions = { toggleHelp: jest.fn() }
  const props = { actions, showHelp: true, darkMode: false }

  beforeEach(() => {
    actions.toggleHelp.mockClear()
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<HelpModal />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    const wrapper = shallow(<HelpModal {...props} />)
    expect(wrapper.find('.help-modal').length).toBe(1)
    expect(wrapper.getElement().props.isOpen).toBeTruthy()
  })

  it('does not render the help modal when showHelp is false', () => {
    const currProps = { ...props, showHelp: false }
    const wrapper = shallow(<HelpModal {...currProps} />)
    expect(wrapper.find('.help-modal').length).toBe(1)
    expect(wrapper.getElement().props.isOpen).toBeFalsy()
  })

  it('closes the modal when the escape key is pressed', () => {
    const wrapper = mount(<HelpModal {...props} />)
    const elem = wrapper.find('.ReactModal__Content')
    elem.getElement().props.onKeyDown({ keyCode: 27, stopPropagation: jest.fn() })
    expect(actions.toggleHelp).toHaveBeenCalledTimes(1)
  })

  it('closes the modal when the close button is clicked', () => {
    const wrapper = shallow(<HelpModal {...props} />)
    wrapper.find('.help-modal-header-close').simulate('click')
    expect(actions.toggleHelp).toHaveBeenCalledTimes(1)
  })
})
