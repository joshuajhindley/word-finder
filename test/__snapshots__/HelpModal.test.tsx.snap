// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the HelpModal component matches the snapshot 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="help-modal "
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={false}
  onRequestClose={[Function]}
  overlayClassName="modal-overlay"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal-content"
  >
    <div
      className="help-modal-header"
    >
      <div
        className="help-modal-header-title"
      >
        <h1>
          Help
        </h1>
      </div>
      <div
        className="help-modal-header-close"
        onClick={[Function]}
      >
        <h1>
          x
        </h1>
      </div>
    </div>
    <div
      className="help-modal-body"
    >
      <p>
        This is a simple tool to help you find the correct word in a Wordle game.
        <br />
        <br />
        There are three rows of input boxes.
        <br />
        The first row are letters in the correct position.
        <br />
        These are the green letters in your Wordle game.
        <br />
        <br />
        The second row are letters that are in the final word, but are in the wrong position.
        <br />
        These are the yellow letters in your Wordle game.
        <br />
        <br />
        The third row are letters that are not in the word.
        <br />
        These are the grey letters in your Wordle game.
        <br />
        New input boxes are added to this row automatically as you add letters.
        <br />
        <br />
        You can use the 'search' button to find the first 500 words that match the letters you've entered and can reset the input boxes by tapping the 'reset' button.
      </p>
    </div>
  </div>
</Modal>
`;
